dojo/tools/azure_security_center/parser.py

```diff
@@ -0,0 +1,91 @@
+        reader = csv.DictReader(io.StringIO(content))
+
+        findings = []
+
+        for row in reader:
+            if 'unhealthy' == row.get('state').lower():
+                subscription_id = row.get('subscriptionId')
+                subscription_name = row.get('subscriptionName')
+                resource_group = row.get('resourceGroup')
+                resource_type = row.get('resourceType')
+                resource_name = row.get('resourceName')
+                # resourceId doesn't give additional information and can be ignored
+                # resource_id = row.get('resourceId')
+                recommendation_id = row.get('recommendationId')
+                recommendation_name = row.get('recommendationName')
+                recommendation_display_name = row.get('recommendationDisplayName')
+                azure_description = row.get('description')
+                remediation_steps = row.get('remediationSteps')
+                severity = row.get('severity')
+                # firstEvaluationDate is where the Security Center was started the first time
+                # firstEvaluationDate = row.get('firstEvaluationDate')
+                status_change_date = row.get('statusChangeDate')
+                controls = row.get('controls')
+                azure_portal_recommendation_link = row.get('azurePortalRecommendationLink')
+                native_cloud_account_id = row.get('nativeCloudAccountId')
+
+                if resource_name == subscription_id:
+                    resource_name = subscription_name
+
+                title = f'{resource_name} - {recommendation_display_name}'
+
+                if controls.startswith('{"'):
+                    controls = controls[controls.find('{"') + 2:controls.find('":')]
+
+                description = "**Recommendation:** " + recommendation_display_name + \
+                    "\n**Resource Name:** " + resource_name + \
+                    "\n**Resource Type:** " + resource_type
+                if resource_group:
+                    description += "\n**Resource Group:** " + resource_group
+                description += "\n**Description:** " + azure_description + \
+                    "\n**Controls:** " + controls + \
+                    "\n**Subscription:** " + subscription_name + \
+                    "\n**Subscription Id:** " + subscription_id
+                if native_cloud_account_id:
+                    description += "\n**Native Cloud Account Id:** " + native_cloud_account_id
+
+                findings.append(Finding(
+                    title=title,
+                    cwe=1032,  # Security Configuration Weaknesses
+                    test=test,
+                    description=description,
+                    severity=severity,
+                    references=azure_portal_recommendation_link,
+                    mitigation=remediation_steps,
+                    date=datetime.strptime(status_change_date[0:10], '%Y-%m-%d').date(),
```

Comment:
The `date` is the first date saw by the tool. So I guess a better candidate for this one is `firstEvaluationDate` attribute of the report. It could lead to some bug with de-duplication but As we are configuring the de-duplication algo to use uniq id from tool is ok.

Reply:
`firstEvaluationDate` is the date when a check for a resource has run the first time. A lot of checks will have the healthy state, so it doesn't give information at which date the check failed for the first time. This is why I think the `status_change_date` might be better, as we import only unhealthy checks.